torchmultimodal/utils/attention.py

```diff
@@ -26,8 +29,13 @@ def get_extended_attention_mask(attention_mask: Tensor) -> Tensor:
         extended_attention_mask = attention_mask[:, None, :, :]
     elif attention_mask.dim() == 2:
         # Provided a padding mask of dimensions [batch_size, seq_length]
-        # - if the model is an encoder, make the mask broadcastable to [batch_size, num_heads, seq_length, seq_length]
-        extended_attention_mask = attention_mask[:, None, None, :]
+        if is_decoder:
+            extended_attention_mask = get_extended_attention_mask_for_decoder(
```

Comment:
I would just keep these two as separate functions and remove this logic, I don't see a reason to tie them together like this